src/Controller/EventsController.php

```diff
@@ -733,16 +734,25 @@ public function rejectAction(Request $request, PDO $db)
 
         $event_id     = $this->getItemId($request);
         $event_mapper = new EventMapper($db, $request);
+        $reason = filter_var($request->getParameter('reason'), FILTER_SANITIZE_STRING);
 
         if (!$event_mapper->thisUserCanApproveEvents()) {
             throw new Exception("You are not allowed to reject this event", Http::FORBIDDEN);
         }
 
-        $result = $event_mapper->reject($event_id, $request->user_id);
+        $result = $event_mapper->reject($event_id, $request->user_id, $reason);
         if (!$result) {
             throw new Exception("This event cannot be rejected", Http::BAD_REQUEST);
         }
 
+        // Only send an email if we provided a reason
+        if (!empty($reason)) {
```

Comment:
Can we avoid using `empty()` here and instead use a stricter verification?

Reply:
Why wouldn't empty be strict enough? The variable will only ever be a string, an empty string, or false.

Reply:
It could also be a `string` consisting of whitespace only.

Reply:
trim() would fix that.